lighthouse-core/report/html/renderer/psi.js

```diff
@@ -81,6 +80,9 @@ function prepareLabData(LHResult, document) {
   const clonedScoreTemplate = dom.cloneTemplate('#tmpl-lh-scorescale', dom.document());
   const scoreScaleEl = dom.find('.lh-scorescale', clonedScoreTemplate);
 
+  const reportUIFeatures = new ReportUIFeatures(dom);
```

Comment:
this class's scope is starting to get scary now 😆 

my trail of verification that this doesn't do anything we don't want it to

- hm, we need an instance of UIFeatures to `addButton` I wonder why it's not static.
- oh it just needs _dom/_document. what else does the constructor do?
- nothing really, `initFeatures` does the heavy lifting.
- why is `reportUIFeatures.json =` needed then? AFAICT it isn't

Reply:
> this class's scope is starting to get scary now 😆

yes, definitely :) it needs a refactor (e.g. `addButton` could just be on `dom` itself or in `performanceCategoryRenderer` if we really want it scoped just to `addMetricsSectionButton`), but since I was the one who suggested not doing #12327 before the 8.0 release I figured this was fair and we can fix it up afterwards :)

Reply:
> * why is `reportUIFeatures.json =` needed then? AFAICT it isn't

this seems right

And we could make `addButton` static now and pass `dom` in (`initFeatures` would just pass it along too), but that might take concurrent DevTools changes?